hazelcast-integration/spring-hibernate-2ndlevel-cache/src/main/java/com/hazelcast/hibernate/springhibernate2lc/Runner.java

```diff
@@ -1,22 +1,112 @@
 package com.hazelcast.hibernate.springhibernate2lc;
 
+import com.hazelcast.core.DistributedObject;
+import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.hibernate.springhibernate2lc.persistence.Book;
 import com.hazelcast.hibernate.springhibernate2lc.persistence.BookRepository;
+import jakarta.persistence.EntityManager;
+import org.hibernate.Session;
+import org.hibernate.stat.Statistics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.boot.CommandLineRunner;
+import org.springframework.cache.CacheManager;
 import org.springframework.stereotype.Component;
+import org.springframework.transaction.support.TransactionTemplate;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
 
 @Component
 class Runner implements CommandLineRunner {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Runner.class);
 
     private final BookRepository repository;
 
-    public Runner(BookRepository repository) {
+    private final HazelcastInstance hazelcastInstance;
+
+    private final TransactionTemplate transactionTemplate;
+
+    private final CacheManager cacheManager;
+
+    private final EntityManager entityManager;
+
+    public Runner(BookRepository repository, HazelcastInstance hazelcastInstance,
+                  TransactionTemplate transactionTemplate, CacheManager cacheManager, EntityManager entityManager) {
         this.repository = repository;
+        this.hazelcastInstance = hazelcastInstance;
+        this.transactionTemplate = transactionTemplate;
+        this.cacheManager = cacheManager;
+        this.entityManager = entityManager;
     }
 
     @Override
     public void run(String... args) throws Exception {
-        repository.save(new Book(42, "foo"));
-        repository.findAll().forEach(System.out::println);
+        Book book1 = new Book();
+        book1.setName("Alice in Wonderland");
+        Book book2 = new Book();
+        book2.setName("");
+
+
+        List<Book> people = List.of(book1, book2);
+        LOGGER.info("Inserting people {}", people);
+
+        // 2 cache puts
+        transactionTemplate.executeWithoutResult(status -> repository.saveAll(people));
```

Comment:
Why TransactionTemplate? Maybe we can get rid of it to make code simpler?

Reply:
I wanted to show that it runs in seperate transactions, but you're right, let's keep it simple: https://github.com/hazelcast/hazelcast-code-samples/pull/576/commits/f4878b3b02a3ea1ab0bbc0ee62d78c09a01f2c60